### ISSUE TITLE
Integrating Firebase Admob with Firebase Remote Config?

### ISSUE THREAD

**Issue by chuakc92** [googleads/googleads-mobile-unity#314]
Is it possible currently to implement both plugins in the same project?

I've been trying to do so the whole day, but my app crashes on startup on my Android folder with the following error:
http://pastebin.com/5GYMdrbd

I'm using Unity 5.3.5f

**Comment by stewartmiles** [comment]
This is cross posted from https://github.com/firebase/quickstart-unity/issues/9#issuecomment-263494671

**Comment by chuakc92** [comment]
Did you mean error logs in Unity or in Android Monitor? 

There were no errors in Unity. 

For Android Monitor, the pastebin link was all the log I could find from this application.

Could you tell me what is the correct step to integrating Firebase Remote Config and Google Mobile Ads so I could test it out for you?

**Comment by stewartmiles** [comment]
You mentioned on the other thread that you tried...

* Resolve Client Jar, which replaces some play-services files with their 10.0.0 variant (not doing so throws a Failed to repackage resources error in Unity when building)
* Importing Google Play Games, which doesn't fix anything

This was after you successfully built without errors and ended up with a crash when loading the application on your Android device?

**Comment by chuakc92** [comment]
Using the quickstart testapp:

1. Imported Firebase and google-services.json
2. Build failed-  **throws a Failed to repackage resources error** 
3. Resolve Client Jar - **Resolve Client Jar, which replaces some play-services files with their 10.0.0 variant**
3. Build successful
4. Everything working perfectly, including Remote Config. 
5. Imported Google Mobile Ads
6. Unity asks if I wanted to delete obsolete Google.IOSResolver, Google.JarResolver, Google.VersionHandler, play-services-resolver v1.2.2.0.
7. Clicked yes
8. Build successful
9. App crashes immediately and throws: http://pastebin.com/5GYMdrbd
10. Resolve Client Jar
11. Build successful
12. App crashes immediately and throws: http://pastebin.com/5GYMdrbd
13. Imported Google Play Games
14. Build successful
15. App crashes immediately and throws: http://pastebin.com/5GYMdrbd

**Comment by rampara** [comment]
This issue is caused by Firebase and AdMob using conflicting Google Play services dependency versions. As a workaround, do the following:

1. Edit the versions of dependencies declared `Assets/Firebase/Editor/AppDeps.cs` from `"9.8+"` to `"LATEST"`. (Lines 43, 44, 45)
2. Delete the following files from the `Assets/Plugins/Android` directory of your Unity project
- `firebase-common-9.8.0`
- `firebase-common-9.8.0.meta`
- `firebase-iid-9.8.0`
- `firebase-iid-9.8.0.meta`
- `firebase-config-9.8.0`
- `firebase-config-9.8.0.meta`

There will updates made to the [Play Services Resolver library](https://github.com/googlesamples/unity-jar-resolver)  to account for conflicting version of dependencies, resolving this issue without the aforementioned workaround.

**Comment by chuakc92** [comment]
Hello rampara,

I did all of those and, unfortunately, it didn't work.

These were my steps:
1. Import Firebase Remote Config into quickstart-unity Remote Config testapp
2. Build successful, but crashes after Unity logo
3. Resolve Client Jars, which replaces some play-services files with their 10.0.1 variant
4. Build Successful and working perfectly
5. Import GoogleMobileAds
6. Build failed - Failed to repackage resources
7. Resolve Jar
8. Deleted:
firebase-common-9.8.0
firebase-common-9.8.0.meta
firebase-iid-9.8.0
firebase-iid-9.8.0.meta
firebase-config-9.8.0
firebase-config-9.8.0.meta
9. Build Successful, but crashes after Unity logo - http://pastebin.com/L6gUTQTd
10. Import Google Play Games
11. Build Successful, but crashes after Unity logo - http://pastebin.com/MifarDMP
12. Resolve Client Jars
13. Deleted:
firebase-common-9.8.0
firebase-common-9.8.0.meta
firebase-iid-9.8.0
firebase-iid-9.8.0.meta
firebase-config-9.8.0
firebase-config-9.8.0.meta
14. Build Successful, but crashes after Unity logo - http://pastebin.com/x7PhhjSk

**Comment by rampara** [comment]
@chuakc92 did you modify the the versions of dependencies declared `Assets/Firebase/Editor/AppDeps.cs` from `"9.8+"` to `"LATEST"`

**Comment by chuakc92** [comment]
Yes, I did.

**Comment by edumlauer** [comment]
@chuakc92 Close Unity, edit any other *Deps.cs you might have in `Assets/Firebase/Editor`, in my case it was AnalyticsDeps.cs and so I edited lines 43 and 44 to `"LATEST"`. 

Delete all files from `Assets/Plugins/Android` with versions prior to 10.0.1.

Reopen Unity and see if this fixes it for you, it did the job for me.

Editing only AppDeps.cs made so it wouldn't crash at launch but then another exception was thrown when I tried to log an event with Firebase Analytics.

**Comment by tuilanhim** [comment]
@edumlauer This is how I do based on your instruction.

1. Keep these "*-unity" lines as default (1.0.0) and change others lines to "LATEST".
2. Turne Assets > Play Services Resolver Settings > Android Resolver > Enable Background Resolution to false
3. Remove all 9.8.0 & 10.0.1 version files
4. Do Assets > Play Services Resolver Settings > Android Resolver > Resolve Client Jars
5. Build for Android and get "Unable to convert classes to dex format" 
6. Remove all auth files
7. Re-build and now I can get remote settings

**Comment by chuakc92** [comment]
Thank you, @edumlauer  and @tuilanhim, I finally got it to work! 

For anyone else facing this problem, here's what I did:

1. Import Firebase Remote Config into quickstart-unity Remote Config testapp
2. Import GoogleMobileAds
3. Resolve Client Jars
4. Went into AppDeps and changed JUST line 43 and 44 from "9.8+" to "LATEST"
5. Deleted:
firebase-common-9.8.0
firebase-common-9.8.0.meta
firebase-iid-9.8.0
firebase-iid-9.8.0.meta
firebase-config-9.8.0
firebase-config-9.8.0.meta
6. Resolve Client Jars
7. Your Android >> Plugin folder should look like this now: http://imgur.com/a/bykBG
8. Build

Remote Config is working now. Going to test Mobile Ads.
